extension/src/project/projectCommands.ts

```diff
@@ -99,6 +100,45 @@ export function registerProjectCommands(context: vscode.ExtensionContext) {
             }
         }));
 
+        context.subscriptions.push(vscode.commands.registerCommand('autolisp.addWorkspaceFile2Project', async (clickedFile: vscode.Uri, selectedFiles: vscode.Uri[]) => {
+            if (!selectedFiles || selectedFiles.length === 0 || getWarnIsSearching()){
+                return;
+            }
+            let selectedDirs = selectedFiles.filter(f => fs.statSync(f.fsPath, { bigint: false}).isDirectory());
+            selectedFiles = selectedFiles.filter(f => fs.statSync(f.fsPath, { bigint: false}).isFile());
+            
+            selectedDirs.forEach(dir => {
+                fs.readdirSync(dir.fsPath).forEach(name => {
+                    const fspath = dir.fsPath.replace(/[\/\\]$/, '') + '\\' + name;
+                    if (fs.existsSync(fspath) && fs.statSync(fspath, { bigint: false}).isFile()) {
+                        if (AutoLispExt.Documents.getSelectorType(fspath) === AutoLispExt.Selectors.lsp){
+                            selectedFiles.push(vscode.Uri.file(fspath));
```

Comment:
Maybe it needs to test if the selectedFiles is null (the workspace has no lisp files)

Reply:
I don't think this is even possible. From what I can tell, the context menu won't show up unless its on/over an LSP file or a Directory. If it is over a directory, then we have at least 1 URI even if it doesn't contain any files. IE, it just won't have anything to do work with.